gemini/getting-started/intro_gemini_2_0_flash_thinking_mode.ipynb

```diff
@@ -0,0 +1,579 @@
+      },
+      "source": [
+        "### **Example 1**: Code simplification\n",
+        "\n",
+        "First, try with a simple code comprehension and simplification example."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "dLhhffx2C9nn"
+      },
+      "outputs": [],
+      "source": [
+        "response = client.models.generate_content(\n",
+        "    model=MODEL_ID,\n",
+        "    contents=\"How can I simplify this? `(Math.round(radius/pixelsPerMile * 10) / 10).toFixed(1);`\",\n",
+        ")\n",
+        "\n",
+        "print(response.candidates[0].content)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "d6cOmdVPC9nn"
+      },
+      "source": [
+        "The model response has multiple parts. While you could use `response.text` to get all of it right away as usual it's actually more interesting to check each of them separately when using the thinking mode.\n",
+        "\n",
+        "The first part is the \"inner thoughts\" of the model, that where it analyzes the problem and comes up with its strategy. Field `thought` indicates if the part is thought from the model."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "c2e4f7b7cc2c"
+      },
+      "outputs": [],
+      "source": [
+        "print(response.candidates[0].content.parts[0].thought)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "OgaDQSL2C9nn"
+      },
+      "outputs": [],
+      "source": [
+        "Markdown(response.candidates[0].content.parts[0].text)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Ig6POLZnC9no"
+      },
+      "source": [
+        "Most of the time you won't need to check the thoughts as you'll be mostly interested in the answer, but having access to them gives you a way to check where the answers come from and how the model comes up with it. It's not a black box anymore!\n",
+        "\n",
+        "Then the second part is the actual answer:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "7cbb7f5d9faf"
+      },
+      "outputs": [],
+      "source": [
+        "print(response.candidates[0].content.parts[1].thought)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "BJ2aYsAhC9no"
+      },
+      "outputs": [],
+      "source": [
+        "Markdown(response.candidates[0].content.parts[1].text)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "3ZqndSMtC9no"
+      },
+      "source": [
+        "As a comparison here's what you'd get with the \"classic\" [Gemini 2.0 Flash](https://cloud.google.com/vertex-ai/generative-ai/docs/gemini-v2) model.\n",
```

Comment:
```suggestion
        "As a comparison, here's what you'd get with the \"classic\" [Gemini 2.0 Flash](https://cloud.google.com/vertex-ai/generative-ai/docs/gemini-v2) model.\n",
```

Reply:
I also might recommend putting the "classic" example first before Thinking Mode examples